File: src/PHP/lua_map_template.php
<?php
#!/usr/bin/env php
function generateMapScript(){
	
	$mapName = (string)readline("Enter the map name ");
	$scriptFolder = '/script';
	$p = 'p';
	$mapFolderName = $p.$mapName;
	$mainMapScript = '_'.$p.$mapName.'.lua';
	
	$stateMapTemplate = <<<STRING
State
{
	StateName = INIT,

	OnOneTimeEvent
	{
		Conditions =
		{
	    	-- fill_me
		},
		Actions =
		{

		},
		GotoState = MAIN,
	},
};

State
{
	StateName = MAIN,

	OnOneTimeEvent
	{
		Conditions =
		{
		    -- fill_me
		},
		Actions =
		{
            -- fill_me
		},
	},

};
STRING;

  mkdir($mapFolderName.$scriptFolder,0777,true);
	$mapFile = fopen($mainMapScript,'w+');
	fwrite($mapFile,$stateMapTemplate);
	fclose($mapFile);
	
	if(is_dir($mapFolderName.$scriptFolder)){
		
		rename(basename($mainMapScript),"$mapFolderName/$scriptFolder/$mainMapScript");
		
	}

}

generateMapScript();

File: src/Python/lua_map_template.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

# Need to think about better option for this template generator.
# I need to add more options for this
import pathlib
import os
import shutil

def generateMapScript():
  map_name = input("Please enter the map name you need\n")
  script_folder = 'script'
  p = 'p'
  map_folder_name = p + map_name
  main_map_script = '_'+ p + map_name + '.lua'
  map_state_template = '''State
{
	StateName = "INIT",

	OnOneTimeEvent
	{
		Conditions =
		{
	    	-- fill_me
		},
		Actions =
		{

		},
		GotoState = "MAIN",
	},
};

State
{
	StateName = "MAIN",

	OnOneTimeEvent
	{
		Conditions =
		{
		    -- fill_me
		},
		Actions =
		{
            -- fill_me
		},
	},

};\n
'''
  os.makedirs(f'{map_folder_name}/{script_folder}')
  
  f = open(main_map_script,'w')
  f.write(map_state_template)
  f.close()
  
  shutil.move(os.path.basename(main_map_script),f'{map_folder_name}/{script_folder}')
  
generateMapScript()
